EGO Forum Enhancement.user.js

```diff
@@ -522,6 +523,91 @@ function handleApplicationPage() {
+            var awardId = awardImageClasses.substring(awardImageClasses.indexOf('--') + 2);
+            createButton("/award-system/" + awardId + "/recent?username=" + username, "Find Issue Reason" , contentDiv.querySelector('.contentRow-main'), '_blank', true);
+        });
+    });
+}
+
+/**
+ * Adds a form to the recent awards page to find a user's award
+ * @returns void
+ */
+function handleRecentAwardPage() {
+    var url = window.location.href;
+    var awardId = url.substring(url.indexOf('/award-system/') + 14, url.indexOf('/recent'));
+    var form = document.createElement('form');
+    form.setAttribute('action', '/award-system/' + awardId + '/recent');
+    form.setAttribute('method', 'get');
+    form.setAttribute('class', 'block');
+    form.innerHTML = '<div class="block-container"><h3 class="block-minorHeader">Find User Award</h3><div class="block-body block-row"><input type="text" class="input" data-xf-init="auto-complete" data-single="true" name="username" data-autosubmit="true" maxlength="50" placeholder="Name…" autocomplete="off"></div></div>';
+    var pageWrapper = document.querySelector('.p-pageWrapper');
+    pageWrapper.insertBefore(form, pageWrapper.querySelector('.p-body'));
+}
+
+/**
+ * Directs the url to the page containing the award of the username that was supplied
+ * @returns void
+ */
+function handleFindAwardPage() {
+    var url = window.location.href;
+    var awardId = url.substring(url.indexOf('/award-system/') + 14, url.indexOf('/recent'));
+    var userToFind = url.substring(url.indexOf('username=') + 9);
+    var maxPagesA = document.querySelector('.pageNav-main > :last-child > a');
+    if(maxPagesA != null) {
+        var maxPages = maxPagesA.innerHTML;
+    }
+    else {
+        var maxPages = 1;
+    }
+    for(var i = 1; i <= maxPages; i++) {
+        parseAwardPage(i, userToFind, awardId);
+    }
+}
+
+/**
+ * Internal function to parse a page of awards
+ * @param {number} pageNum the current number of the page being parsed
+ * @param {string} userToFind the username to find
+ * @param {string} awardId the id of the award to find
+ */
+function parseAwardPage(pageNum, userToFind, awardId) {
+    var pageHtml = document.createElement("html");
+    fetch("/award-system/" + awardId + "/recent?page=" + pageNum).then(function (response) {
+        response.text().then(result = function (text) {
+            pageHtml.innerHTML = text;
+            var bodyDiv = pageHtml.querySelector('.block-body');
+            Array.from(bodyDiv.children).forEach(function (div) {
+                if(div.getAttribute('data-author') == userToFind) {
+                    window.location.href = "/award-system/" + awardId + "/recent?page=" + pageNum + "&spotlight=" + userToFind;
+                }
+            });
+        });
+    });
+}
+
+function handleAwardSpotlight() {
+    var url = window.location.href;
+    var userToFind = url.substring(url.indexOf('spotlight=') + 10);
```

Comment:
This would be better done with Regex.

```suggestion
    var userToFind = url.match(/^https:\/\/www\.edgegamers\.com\/award-system\/\d+\/recent\?page=\d+\&spotlight=(.+)$/)[1];
```